### PR TITLE
Add comparison

### LINKED ISSUE #25: Add framework comparison
People often ask how a new framework compares to similar solutions. It would be useful to provide a high level comparison with current popular solutions

### PR DESCRIPTION
**Pull request by lhorie**
Resolves #25

### CODE REVIEW

File: docs/guides/framework-comparison.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Framework comparison
```
Comment (mlmorg) [inline docs/guides/framework-comparison.md]: next.js?
create-react-app?
electrode?

Comment (lhorie) [reply docs/guides/framework-comparison.md]: Added comparisons for those 3

File: docs/guides/framework-comparison.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+In addition to these core areas, the FusionJS team provides several plugins for various things: from React and Redux to translations and CSRF protection. This makes FusionJS a one-stop shop for full stack development, while at the same time making it easy to keep bundle size small by allowing developers to only include what they need.
+
+Here is how FusionJS compares to some popular frameworks and libraries:
+
+### React
+
+[React](https://reactjs.org/) is a popular and mature library developed by Facebook for implementing component-based UIs. FusionJS can be used seamlessly with React: JSX, ES2017, hot module reloading, server-side rendering, etc all work out of the box. In addition, the FusionJS core is view-layer agnostic and we are planning on adding plugins for Preact in the future.
+
+Through plugins, FusionJS provides various extra features on top of vanilla React: it provides an easy way to do code splitting, it supports async server-side rendering, it allows plugins to automatically installs providers when needed, etc.
```
Comment (angus-c) [inline docs/guides/framework-comparison.md]: s/installs/install

File: docs/guides/framework-comparison.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+#### create-react-app
+
+[create-react-app](https://github.com/facebookincubator/create-react-app) is a CLI that scaffolds a React project, as well as configure Webpack, Babel, Eslint, Jest, etc. It does not support some things out of the box (most notably, [server rendering and hot reloading of components](https://github.com/facebookincubator/create-react-app#limitations)), but it provides the ability to "eject", so you can use it as a boilerplate generator and customize the build workflow from there.
```
Comment (angus-c) [inline docs/guides/framework-comparison.md]: >> It does not support some things out of the box

There are some things it doesn't support out of the box

File: docs/guides/framework-comparison.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+FusionJS is similar to create-react-app in the sense that it also provides a Webpack/Babel setup, but in addition to that, it also provides higher level tools like server-side rendering, more powerful code splitting and fully integrated hot reloading. FusionJS also provides a modern HTTP server API (through [Koa](http://koajs.com/)) and an universal code environment.
```
Comment (angus-c) [inline docs/guides/framework-comparison.md]: > it also provides

it provides

File: docs/guides/framework-comparison.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+The main difference is that Next.js focuses on React development and doesn't provide facilities for developing full-on backends, whereas FusionJS is a full stack framework, which supports backend development through [Koa](http://koajs.com/) middlewares and RPC plugins.
+
+Both Next.js and FusionJS can be used with a variety of 3rd party libraries, but in the interest of avoiding analysis paralysis, the FusionJS team also provides a curated set of plugins that we consider important for production-quality applications and that can integrate into a base app in a way that is performant and boilerplate-free.
+
+### Electrode.io
+
+[Electrode](http://www.electrode.io/) is a framework developed by Walmart Labs designed for building server-rendered React applications. It's very similar in scope to FusionJS, and offers many of the same features (e.g. server rendering, hot reloading, etc).
+
+Electrode has been around for longer and has more mature tooling surrounding bundle analysis and deployment. FusionJS is a much younger project and we plan on focusing more on automating performance optimization strategies and solidifying features related to application health monitoring.
```
Comment (angus-c) [inline docs/guides/framework-comparison.md]: s/for longer/longer

File: docs/guides/what-is-fusion.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# What is FusionJS
+
+FusionJS is a web application framework developed by Uber. Because Uber operates on a large scale in countries with slow mobile networks, performance is a big driving factor for FusionJS.
+
+It has a modular architecture in order to promote small bundle sizes, and it is designed in such a way that performance milestones in FusionJS' development pipeline can be rolled out to consumers of the framework via version bumps, as opposed to requiring big migrations or offloading that work to app developers altogether.
```
Comment (angus-c) [inline docs/guides/what-is-fusion.md]: > offloading that work to app developers altogether.

entirely offloading that work to app developers

**Comment by lhorie** [comment]
!merge

**Pull request merged**